sdk/core/System.ClientModel/src/Convenience/ConnectionCollection.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Represents a collection of client connections.
+/// </summary>
+[JsonConverter(typeof(ConnectionCollectionConverter))]
+[DebuggerTypeProxy(typeof(ConnectionCollectionViewer))]
+public class ConnectionCollection : KeyedCollection<string, ClientConnection>
+{
+    /// <summary>
+    /// Gets the key for a given <see cref="ClientConnection"/>.
+    /// </summary>
+    protected override string GetKeyForItem(ClientConnection item) => item.Id;
+
+    /// <summary>
+    /// Adds a range of <see cref="ClientConnection"/> instances to the collection.
+    /// </summary>
+    /// <param name="connections">The connections to add.</param>
+    internal void AddRange(IEnumerable<ClientConnection> connections)
+    {
+        foreach (ClientConnection connection in connections)
+        {
+            Add(connection);
+        }
+    }
+}
```

Comment:
@KrzysztofCwalina Should we expose this method as public?  

It's being used in the `ProjectClient` [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cloudmachine/Azure.CloudMachine/src/ProjectClient.cs#L85).

Reply:
yes. let's do